### ISSUE TITLE
Issues with self signed Certificates in Dev Environments

### ISSUE THREAD

**Issue by pdimitri** [emiloberg/liferay-ddmtool#9]
I'm unable to use your great Tool in our Project Setup, because all our Servers use SSL Certficates. On our Dev Servers with self signed Certificates I see this Error Message:

Getting data from server
Downloading id's

Error
Unknown error: {"code":"DEPTH_ZERO_SELF_SIGNED_CERT"}

On the Prod Server, which uses an official  signed Cetificate I see this one:

Getting data from server
Downloading id's

Error
Unable to check CSRF token

Any Ideas on how we could use the Tool with secured Connections?

**Comment by emiloberg** [comment]
Hi,

Are you using an old version/config files?

Take a look at this issue, does this help? https://github.com/emiloberg/liferay-ddmtool/issues/7

**Comment by pdimitri** [comment]
You're right. I was using an older version, since I'm using your great tool until you presented it in DevCon last year. Unfortunately it still won't work in my current environment. Setting the "allowSelfSignedCerts" parameter removed the "DEPTH_ZERO_SELF_SIGNED_CERT" but now I get an "java.lang.IllegalStateException". On the production server (with the officially signed certificate) there's still the "Unable to check CSRF token" Error.

Sorry for bothering you, but I'd really like to use your tool again in my current project as we massively use ddm stuff in that project.

**Comment by emiloberg** [comment]
Am I understanding you correctly if I say that:
- On your stage environment, where you previously got `DEPTH_ZERO_SELF_SIGNED_CERT`, you're now - after updating the tool itself and updating your config files - getting `java.lang.IllegalStateException`
- On your production environment you're getting `Unable to check CSRF token`

Correct?

Will you be joining DevCon this year?

**Comment by pdimitri** [comment]
Correct!

BTW: When I install you Tool using “npm install –g “ there are lots of
missing modules which I had to install manually to make it work, like
minimist, q, cli, cli-color, node-print, fs-extra and may others.

Maybe we’ll find a view minutes next week in Darmstadt, then I may show you
my setup. I’ll have my Laptop with me at our stand on LPSF and DevCon.

Kind regards

Peter

_Von:_ Emil Öberg [mailto:notifications@github.com]
_Gesendet:_ Freitag, 2. Oktober 2015 10:01
_An:_ emiloberg/liferay-ddmtool
_Cc:_ pdimitri
_Betreff:_ Re: [liferay-ddmtool] Issues with self signed Certificates in
Dev Environments (#9)

Am I understanding you correctly if I say that:
- On your stage environment, where you previously got
  DEPTH_ZERO_SELF_SIGNED_CERT, you're now - after updating the tool itself
  and updating your config files - getting java.lang.IllegalStateException
- On your production environment you're getting Unable to check CSRF
  token

Correct?

Will you be joining DevCon this year?

**Comment by emiloberg** [comment]
Cool!

Let's solve this as DevCon! Grab a hold of me if I don't get to you first!
